### ISSUE TITLE
IndexOutOfRangeException when starting solution with 3 Node.js projects

### ISSUE THREAD

**Issue by nearwood** [Microsoft/nodejstools#909]
Hello again,

```
Microsoft Visual Studio Professional 2015
Version 14.0.25123.00 Update 2
Microsoft .NET Framework
Version 4.6.01055
...
JavaScript Language Service   2.0
JavaScript Project System   2.0
Node.js Tools   1.2.40329.01
Node.js Tools - Profiling   1.2.40329.01
TypeScript   1.8.30.0
VSTestHost   1.0.1
```

I was running 1.2 Alpha and ran into a System.IndexOutOfRangeException:

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IndexOutOfRangeException
   at Microsoft.NodejsTools.SourceMapping.SourceMap..ctor(System.IO.TextReader)
   at Microsoft.NodejsTools.SourceMapping.SourceMapper.TryGetMapInfo(System.String)
   at Microsoft.NodejsTools.SourceMapping.SourceMapper.MapToOriginal(System.String)
   at Microsoft.NodejsTools.SourceMapping.SourceMapper.GetOriginalFileName(System.String, System.Nullable`1<Int32>, System.Nullable`1<Int32>)
   at Microsoft.NodejsTools.Debugger.NodeDebugger.GetOrAddModule(Microsoft.NodejsTools.Debugger.NodeModule, Microsoft.NodejsTools.Debugger.NodeModule ByRef, Microsoft.NodejsTools.Debugger.NodeStackFrame)
   at Microsoft.NodejsTools.Debugger.NodeDebugger.AddModules(System.Collections.Generic.IEnumerable`1<Microsoft.NodejsTools.Debugger.NodeModule>)
   at Microsoft.NodejsTools.Debugger.NodeDebugger.OnCompileScriptEvent(System.Object, Microsoft.NodejsTools.Debugger.Communication.CompileScriptEventArgs)
   at Microsoft.NodejsTools.Debugger.Communication.DebuggerClient.HandleEventMessage(Newtonsoft.Json.Linq.JObject)
   at Microsoft.NodejsTools.Debugger.Communication.DebuggerClient.OnOutputMessage(System.Object, Microsoft.NodejsTools.Debugger.Communication.MessageEventArgs)
   at Microsoft.NodejsTools.Debugger.Communication.DebuggerConnection+<ReceiveAndDispatchMessagesWorker>d__25.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_1(System.Object)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

I took down the code and I believe I fixed the immediate problem, but I am not familiar with the codebase so it may be insufficient. I will submit a PR shortly.

**Comment by nearwood** [comment]
I'm not sure what, in my particular setup, caused the issue. I also cannot share any proprietary items. I can provide censored package.json files though, for a start:

Project A
```
{
  "dependencies": {
    "body-parser": "^1.15.0",
    "compression": "^1.6.1",
    "connect-redis": "^3.0.2",
    "connect-sequelize": "^2.0.2",
    "cookie-parser": "^1.4.1",
    "express": "^4.13.4",
    "express-session": "^1.13.0",
    "mongoose": "^4.4.13",
    "passport": "^0.3.2",
    "passport-local": "^1.0.0",
    "redis": "^2.6.0-1",
    "sequelize": "^3.22.0",
    "vash": "^0.11.3"
  }
}
```

Project B
```
{
  "dependencies": {
    "async-props": "^0.3.2",
    "async-waterfall": "^0.1.5",
    "compression": "^1.6.1",
    "express": "^4.13.3",
    "fast-json-patch": "^0.5.6",
    "fs-extra": "^0.26.7",
    "iso-http": "0.0.5",
    "jquery": "^1.12.2",
    "jsonpatch-to-mongodb": "^0.2.0",
    "material-ui": "^0.15.0-alpha.2",
    "mongodb": "^2.1.4",
    "ramda": "^0.20.0",
    "react": "^0.14.7",
    "react-addons-transition-group": "^0.14.7",
    "react-dom": "^0.14.7",
    "react-router": "^2.0.1",
    "react-tap-event-plugin": "^0.2.2",
    "rest": "^1.3.2",
    "roboto-fontface": "^0.4.5",
    "serve-favicon": "^2.3.0",
    "socket.io": "^1.4.4",
    "socket.io-client": "^1.4.5"
  },
  "devDependencies": {
    "babel-core": "^6.7.4",
    "babel-loader": "^6.2.4",
    "babel-preset-es2015": "^6.6.0",
    "babel-preset-react": "^6.5.0",
    "css-loader": "^0.23.1",
    "extract-text-webpack-plugin": "^1.0.1",
    "file-loader": "^0.8.5",
    "npm-run-all": "^1.7.0",
    "style-loader": "^0.13.1",
    "webpack": "^1.12.14",
    "gulp": "^3.9.1"
  }
}
```

Project C
```
{
  "dependencies": {
    "express": "^4.13.4",
    "http-proxy": "^1.13.2",
    "serve-favicon": "^2.3.0"
  }
}
```

**Comment by mjbvz** [comment]
Fixed by #909
